src/transport/SecureSessionMgr.cpp

```diff
@@ -184,13 +181,8 @@ CHIP_ERROR SecureSessionMgr::SendMessage(SecureSessionHandle session, PayloadHea
         SuccessOrExit(err);
     }
 
-    // The start of buffer points to the beginning of the encrypted header, and the length of buffer
-    // contains both the encrypted header and encrypted data.
-    // Locally store the start and length of the retained buffer after accounting for the size of packet header.
-    headerSize = packetHeader.EncodeSizeBytes();
-
-    msgStart = static_cast<uint8_t *>(msgBuf->Start() - headerSize);
-    msgLen   = static_cast<uint16_t>(msgBuf->DataLength() + headerSize);
+    err = packetHeader.EncodeBeforeData(msgBuf);
```

Comment:
It's out of range, so I can't make a suggestion on it, but `SecureSessionMgr::SendEncryptedMessage` now has an incorrect comment.  It says "the transport will attach the packet header on top of it" when it should say "SendMessage will attach the packet header on top of it".